src/corelibs/U2View/src/ov_msa/MultilineScrollController.cpp

```diff
@@ -177,11 +177,58 @@ int MultilineScrollController::getViewHeight() {
     return childrenScrollArea->height();
 }
 
+bool MultilineScrollController::checkBoundary() {
+    int firstBase = -1;
+    int prevFistBase = -1;
+    bool needUpdate = false;
+    int childrenCount = ui->getChildrenCount();
+    int alignmentLen = maEditor->getAlignmentLen();
+    for (int i = 0; i < childrenCount; i++) {
+        firstBase = ui->getUI(i)->getScrollController()->getFirstVisibleBase(false);
+        if (firstBase == prevFistBase) {
+            needUpdate = true;
+            break;
+        }
+        if ((firstBase + 1) >= alignmentLen) {
+            needUpdate = true;
+            break;
+        }
+        prevFistBase = firstBase;
+    }
+    if (needUpdate) {
+        const int length = ui->getSequenceAreaBaseLen(0);
```

Comment:
nit: do not use const for locals (you use it inconsistently across the patch and UGENE code (99%) does not use it at all

Reply:
fixed